### PR TITLE
Include job count in WorkerStatus and load_fnc

### PR DESCRIPTION
**Pull request by keepingitneil**

### CODE REVIEW

**Comment by theomonnom** [comment]
I think we should pass the Worker instance to the load_fnc, not juste the number of jobs bc it seems very specific

**Comment by keepingitneil** [comment]
ahh missed this, ok yeah that makes sense - will update

File: livekit-agents/livekit/agents/worker.py [hunk 1]
```diff
@@ -683,15 +696,29 @@ def _on_process_job_launched(self, proc: ipc.job_executor.JobExecutor) -> None:
...
-            None, self._opts.load_fnc
-        )
+        def load_fnc():
+            signature = inspect.signature(self._opts.load_fnc)
+            parameters = signature.parameters
+            accepts_kwargs = any(
+                param.kind == inspect.Parameter.VAR_KEYWORD
+                for param in parameters.values()
+            )
+            if not accepts_kwargs and "job_cnt" not in parameters:
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/worker.py:715]: We should not get the argument by name,  I don't think the argument should be a kwarg, but the worker would be the first positional one. I don't think we will add any other variables in the future.

File: livekit-agents/livekit/agents/worker.py [hunk 2]
```diff
@@ -59,6 +68,10 @@ async def _default_request_fnc(ctx: JobRequest) -> None:
     await ctx.accept()
 
 
+class LoadFunction(Protocol):
+    def __call__(self, worker: Worker) -> float: ...
+
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/worker.py:73]: why not `Callable[[Worker], float]`?

File: livekit-agents/livekit/agents/worker.py [hunk 3]
```diff
@@ -690,15 +703,24 @@ def _on_process_job_launched(self, proc: ipc.job_executor.JobExecutor) -> None:
...
             return
 
-        current_load = await asyncio.get_event_loop().run_in_executor(
-            None, self._opts.load_fnc
-        )
+        def load_fnc():
+            signature = inspect.signature(self._opts.load_fnc)
+            parameters = list(signature.parameters.values())
+            if len(parameters) == 0:
+                return self._opts.load_fnc()  # type: ignore
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/worker.py:719]: not sure about the `type: ignore` here
What are they used for?

File: livekit-agents/livekit/agents/ipc/proc_pool.py [hunk 4]
```diff
@@ -66,6 +66,9 @@ def get_by_job_id(self, job_id: str) -> JobExecutor | None:
             None,
         )
 
+    def get_running_jobs(self) -> list[RunningJobInfo]:
+        return [x.running_job for x in self._executors if x.running_job]
+
     def start(self) -> None:
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/ipc/proc_pool.py:72]: You can use https://github.com/livekit/agents/blob/0c9eac91938d21ba4c2f2d16cfe05fb8fa237786/livekit-agents/livekit/agents/worker.py#L303 instead

File: livekit-agents/livekit/agents/worker.py (hunk 3 above)
Comment (keepingitneil) [reply livekit-agents/livekit/agents/worker.py:717]: type:ignore is needed because it's not known which load_fnc we have, one with arg or one without?

**Review by theomonnom — APPROVED** [review]
+ changesets otherwise lgtm

File: livekit-agents/livekit/agents/worker.py (hunk 2 above)
Comment (keepingitneil) [reply livekit-agents/livekit/agents/worker.py]: changed it, thx

File: livekit-agents/livekit/agents/ipc/proc_pool.py [hunk 5]
```diff
@@ -66,7 +66,7 @@ def get_by_job_id(self, job_id: str) -> JobExecutor | None:
             None,
         )
 
-    def get_running_jobs(self) -> list[RunningJobInfo]:
```
Comment (theomonnom) [inline livekit-agents/livekit/agents/ipc/proc_pool.py]: lol what I mean is that this function is already in the worker

Comment (keepingitneil) [reply livekit-agents/livekit/agents/ipc/proc_pool.py]: ohhhh lol removed

**Review by theomonnom — APPROVED** [review]

**Pull request merged**